src/main.rs

```diff
@@ -0,0 +1,41 @@
+// Write a CLI program that call the bash file rails-new inside the bin folder.
+
+// use std::process::Command;
+mod docker_client;
+mod rails_new;
+use rails_new::Cli;
+use std::io::Write;
+
+use clap::Parser;
+
+use crate::docker_client::DockerClient;
+
+fn main() {
+    let cli = Cli::parse();
+
+    // read the content of the DOCKERFILE and store it in a variable
+    let dockerfile = include_str!("../Dockerfile");
+
+    let ruby_version = cli.ruby_version.unwrap();
+    let rails_version = cli.rails_version.unwrap();
+
+    // Run docker build --build-arg RUBY_VERSION=$RUBY_VERSION --build-arg RAILS_VERSION=$RAILS_VERSION -t rails-new-$RUBY_VERSION-$RAILS_VERSION
+    // passing the content of DOCKERFILE to the command stdin
+    let mut child = DockerClient::build_image(&ruby_version, &rails_version)
+        .spawn()
+        .expect("Failed to execute process");
+
+    let stdin = child.stdin.as_mut().expect("Failed to open stdin");
+    stdin.write_all(dockerfile.as_bytes()).unwrap();
```

Comment:
If you only want this as bytes, you can `include_bytes!(...)` instead of `include_str!(...).as_bytes()`